File: animals/AnimalModel.php
<?php

class AnimalModel
{
    public string|null $id;
    public string $name;
    public string $birth_date;
    public string $specie;
    public string $farmer;
    public string $status;
    public string $picture;

    public function __construct($name, $birth_date, $specie, $farmer, $status, $picture, $id = null)
    {
        $this->id = $id;
        $this->name = $name;
        $this->birth_date = $birth_date;
        $this->specie = $specie;
        $this->farmer = $farmer;
        $this->status = $status;
        $this->picture = $picture;
    }
}

File: farm/src/main/java/farmhub/service/models/AnimalModel.java
package farmhub.service.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AnimalModel {
    @JsonProperty("id")
    public String id;

    @JsonProperty("name")
    public String name;

    @JsonProperty("birth_date")
    public String birthDate;

    @JsonProperty("specie")
    public SpecieModel specie;

    @JsonProperty("farmer")
    public FarmerModel farmer;

    @JsonProperty("status")
    public String status;

    @JsonProperty("picture")
    public PictureModel picture;
}
